File: Programacion II/Participacion/Circulo2D.java
public class Circulo2D {
	private double x,y;
	private double radio;
	
	public double getRadio() {
		return radio;
	}
	
	public double getX() {
		return x;
	}
	
	public double getY() {
		return y;
	}
	
	public Circulo2D() {
		this.x=0;
		this.y=0;
		this.radio=1;
	}
	
	public Circulo2D(double x,double y,double radio) {
		this.x=x;
		this.y=y;
		this.radio=radio;
	}
	
	public double getArea() {
		return Math.PI*Math.pow(this.radio,2);
	}
	
	public double getPerimetro() {
		return 2*this.radio*Math.PI;
	}
	
	public boolean contiene(double x,double y) {
		double distancia=Math.sqrt(Math.pow((x-this.x),2)+Math.pow((y-this.y),2));
		if(distancia<this.radio) {
			return true;
		}else {
			return false;
		}
	}
	
	public boolean contiene(Circulo2D circulo) {
		double distancia=Math.sqrt(Math.pow((circulo.x-this.x),2)+Math.pow((circulo.y-this.y),2));
		if(distancia<(this.radio-circulo.radio)) {
			return true;
		}else {
			return false;
		}
		
	}
	public boolean sobrepone(Circulo2D circulo) {
		double distancia=Math.sqrt(Math.pow((circulo.x-this.x),2)+Math.pow((circulo.y-this.y),2));
		if(distancia<(this.radio+circulo.radio)) {
			return true;
		}else {
			return false;
		}
	}

}

File: Programacion II/Participacion/Circulo2D.py
import math
from multimethod import multimethod
class Circulo2D:
    def __init__(self, x=0.0, y=0.0, radio=1.0):
        self._x = x
        self._y = y
        self._radio = radio
    def getX(self):
        return self._x
    def getY(self):
        return self._y
    def get_radio(self):
        return self._radio  
    
    def get_area(self):
        return math.pi * self._radio ** 2

    def get_perimetro(self):
        return 2 * math.pi * self._radio
    @multimethod
    def contiene(self, x:float, y:float):
        distancia = math.sqrt((x - self._x) ** 2 + (y - self._y) ** 2)
        if(distancia < self._radio):
            return True
        return False
    @multimethod
    def contiene(self, otro):
        distancia = math.sqrt((otro._x - self._x) ** 2 + (otro._y - self._y) ** 2)
        if distancia < (self._radio - otro._radio):
            return True
        return False

    def sobrepone(self, otro):
        distancia = math.sqrt((otro._x - self._x) ** 2 + (otro._y - self._y) ** 2)
        if(distancia < (self._radio + otro._radio)):
            return True
        return False


    
c1 = Circulo2D(2, 0, 1)
c2 = Circulo2D(2, 0, 0.5)
c3 = Circulo2D(0, 0, 2)
print(c1.get_area())
print(c1.get_perimetro())

print(c1.contiene(2.5, 0.0))
print(c1.contiene(c2))
print(c1.sobrepone(c3))
